src/models/otp.ts

```diff
@@ -117,7 +117,30 @@ export class OTPEntry implements OTPEntryInterface {
       this.secret = entry.secret;
     }
 
-    this.type = entry.type;
+    // We may have already had some error OTP type entries in the storage
+    // totp = 1
+    // hotp = 2
+    // battle = 3
+    // steam = 4
+    // hex = 5
+    // hhex = 6
+
+    if ((entry.type as unknown) === 1) {
```

Comment:
Entries are stored with a **string** `type`
https://github.com/Authenticator-Extension/Authenticator/blob/6511920bac98576e3c97fe2d213b23e9dc899422/src/models/storage.ts#L225
so why is this `if` block needed?

Reply:
And if you were thinking `entry.type` might be a **string** number such as `"1"`, the `===` comparison won't allow it.

I'm not saying you're wrong, just trying to understand it 🙂

Reply:
Error OTP type entries would only ever be `1`, there is no need to check the other numbers.

[See my comment below](https://github.com/Authenticator-Extension/Authenticator/pull/1283#issuecomment-2378100678)